### PR TITLE
[CRASH-POTENTIAL] Incorrect L1 Regularizer Formula 

### PR DESCRIPTION
**Pull request by ShindeShivam**
**Chapter:12**
**Cell:Other Custom Functions**

###  Important Fix: Incorrect L1 Regularizer Formula :-

The current implementation of the `L1Regularizer` incorrectly applies the regularization factor *before* taking the absolute value of the weights:

`return tf.reduce_sum(tf.abs(self.factor * weights))`

**But the actual L1 Regularizer Formula is:**
###L1 loss = λ⋅∑∣ wi ∣

**So the correct version of this is:**

`return self.factor * tf.reduce_sum(tf.abs(weights))`

### Transparency Note:
The word "CRASH" in the title is used purely to grab your attention — because my last few PRs haven't been reviewed yet, possibly due to your busy schedule
I just wanted to make sure this particular fix — which impacts correctness — doesn't slip through unnoticed.

### Pending Pull Requests
Here are a few other PRs I’ve opened recently that are still awaiting review:

#196
#202 
#205 
#206
#207

### CODE REVIEW

**Comment by ShindeShivam** [comment]
Thanks for clarifying.
I agree that for a positive scalar factor both implementations are mathematically equivalent.
My suggestion was based on the canonical L1 formula (λ * Σ|wᵢ|) and also I just checked that Keras’ own L2 regularizer follows this same pattern applying the factor after the reduction for clarity and consistency.
So while both work here, this version matches the style used in Keras and the standard definition.

Screenshot_20250810_163002_Chrome
